app/src/test/kotlin/compiler/intermediate_form/ControlFlowGraphBuilderTest.kt

```diff
@@ -0,0 +1,144 @@
+package compiler.intermediate_form
+import compiler.common.reference_collections.referenceHashMapOf
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
```

Comment:
Maybe we could test that methods that add link from **all** final trees actually do it. All current tests have only one final tree, so methods linking only one arbitrary tree instead of all would pass the tests.